packages/react/src/utils/useEventCallback.ts

```diff
@@ -1,4 +1,41 @@
 'use client';
-import { useEffectEvent } from '@floating-ui/react/utils';
+import * as React from 'react';
+import { useLazyRef } from './useLazyRef';
 
-export const useEventCallback = useEffectEvent;
+// https://github.com/mui/material-ui/issues/41190#issuecomment-2040873379
+const useInsertionEffect =
+  (React as any)[`useInsertionEffect${Math.random()}`.slice(0, 5)] || ((fn: any) => fn());
```

Comment:
The slice needs to match the length of the string, but I think we should just go with `{...React}` instead? It's a bit more costly but close to negligible (at least this is used in Floating UI).

Reply:
I have updated to slice away the number with a negative index. I have seen to many `...` in the codebase, I'm trying to stay away from them.